crates/node/builder/src/rpc.rs

```diff
@@ -712,6 +712,8 @@ where
         >,
     >,
     EV: EngineValidatorBuilder<N>,
+    for<'de> <<N as FullNodeTypes>::Types as NodeTypesWithEngine>::Engine:
+        serde::Serialize + serde::Deserialize<'de>,
```

Comment:
hmm, but I'd prefer to keep this on the trait instead, why was this change necessary?

Reply:
sure, done here https://github.com/paradigmxyz/reth/pull/13995/commits/afa91ac281e866ba5fcdb6e10c7d824e62864bcd